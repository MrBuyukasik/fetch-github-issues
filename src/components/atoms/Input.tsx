import React from 'react';
import {TextInput} from 'react-native';
import styled from 'styled-components/native';

import {color} from '../../common/color';

const StyledInput = styled(TextInput)`
  width: 100%;
  background-color: ${color.WHITE};
  border: 2px solid ${color.BLACK};
  margin: 2px;
  padding-left: 8px;
  padding: 10px;
  height: 50px;
  border-width: 1px;
  border-radius: 10px;
  font-size: 18px;
`;

interface Props {
  defaultValue?: string;
  onUpdate(value: string): void;
}

const Input = (props: Props) => {
  const {defaultValue, onUpdate} = props;

  const onChange = (data: string) => {
    onUpdate(data);
  };

  return <StyledInput defaultValue={defaultValue} onChangeText={onChange} />;
};

export default Input;
