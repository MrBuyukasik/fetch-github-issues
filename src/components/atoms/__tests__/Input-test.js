import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';
fetch = jest.fn(() => Promise.resolve());

describe('Input Tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
