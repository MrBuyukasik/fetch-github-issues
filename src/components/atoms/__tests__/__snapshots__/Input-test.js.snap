// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Tests renders correctly 1`] = `
<TextInput
  onChangeText={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#000000",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 18,
        "height": 50,
        "marginBottom": 2,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 2,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": "100%",
      },
    ]
  }
/>
`;
